### PR TITLE
Replace snake_case in rct1 folder

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

**Comment by Gymnasiast** [comment]
This appears to be causing a crash due to an incorrect peep state. Do not merge yet.

File: src/openrct2/rct1/RCT1.h [hunk 1]
```diff
@@ -437,174 +437,174 @@ namespace RCT1
...
         {
             struct
             {
-                uint8_t current_car;  // 0x6B
-                uint8_t current_seat; // 0x6C
+                uint8_t CurrentCar;  // 0x6B
+                uint8_t CurrentSeat; // 0x6C
             };
-            uint16_t time_to_sitdown; // 0x6B
+            uint16_t TimeTo_sitdown; // 0x6B
```
Comment (Gymnasiast) [inline src/openrct2/rct1/RCT1.h:489]:
```suggestion
            uint16_t TimeToSitdown; // 0x6B
```

File: src/openrct2/rct1/RCT1.h [hunk 2]
```diff
@@ -437,174 +437,174 @@ namespace RCT1
...
             struct
             {
-                uint8_t time_to_stand;  // 0x6B
-                uint8_t standing_flags; // 0x6C
+                uint8_t TimeTo_stand;  // 0x6B
```
Comment (Gymnasiast) [inline src/openrct2/rct1/RCT1.h:492]:
```suggestion
                uint8_t TimeToStand;  // 0x6B
```

File: src/openrct2/rct1/RCT1.h [hunk 3]
```diff
@@ -437,174 +437,174 @@ namespace RCT1
...
-        money32 cash_in_pocket;                             // 0xA0
-        money32 cash_spent;                                 // 0xA4
-        int32_t park_entry_time;                            // 0xA8
-        int8_t rejoin_queue_timeout;                        // 0xAC
-        RCT12RideId previous_ride;                          // 0xAD
-        uint16_t previous_ride_time_out;                    // 0xAE
-        RCT12PeepThought thoughts[Limits::MaxPeepThoughts]; // 0xB0
-        uint8_t pad_C4;
+        // WindowPeepRidesUpdate for how to use.
+        uint32_t Id;                                        // 0x9C
```
Comment (Gymnasiast) [inline src/openrct2/rct1/RCT1.h:523]: ID

File: src/openrct2/rct1/RCT1.h [hunk 4]
```diff
@@ -614,174 +614,174 @@ namespace RCT1
...
+        uint16_t Month;
+        uint16_t Day;
+        uint32_t Ticks;
+        uint32_t RandomA;
+        uint32_t RandomB;
+        RCT12TileElement TileElements[Limits::MaxTileElements];
+        uint32_t UnkCounter;
+        Entity Sprites[Limits::MaxEntities];
+        uint16_t NextSpriteIndex;
+        uint16_t FirstVehicleSpriteIndex;
```
Comment (Gymnasiast) [inline src/openrct2/rct1/RCT1.h:626]: entity

File: src/openrct2/rct1/RCT1.h [hunk 5]
```diff
@@ -614,174 +614,174 @@ namespace RCT1
...
+        uint32_t AvailableRides[8];
+        uint32_t AvailableVehicles[8];
+        uint32_t RideFeature1[128];
+        uint32_t RideFeature2[128];
+        uint16_t GuestsInPark;
+        uint16_t Unk198C9E;
+        money32 Expenditure[Limits::ExpenditureTableMonthCount][Limits::ExpenditureTypeCount];
+        uint32_t GuestsInPark2;
+        uint8_t Unk199024;
+        colour_t HandimanColour;
```
Comment (Gymnasiast) [inline src/openrct2/rct1/RCT1.h:658]: handyman

File: src/openrct2/rct1/RCT1.h [hunk 6]
```diff
@@ -614,174 +614,174 @@ namespace RCT1
...
+        uint32_t FinanceChecksum3;
+        uint32_t ScenarioSlotIndexChecksum;
+        char ScenarioWinner[32];
+        uint32_t FinanceChecksum2;
+        char CopyrightNotice[40];
+        uint16_t CheatDetectionSV6SC4[4];
+        uint16_t Unk199C84;
+        uint16_t Unk199C86;
+        uint16_t MapSizeUnits;
+        uint16_t MapSize_Unkb;
```
Comment (Gymnasiast) [inline src/openrct2/rct1/RCT1.h:733]:
```suggestion
        uint16_t MapSizeUnkB;
```

File: src/openrct2/rct1/RCT1.h [hunk 7]
```diff
@@ -437,174 +437,174 @@ namespace RCT1
...
+        StringId NameStringID;     // 0x22
+        uint16_t NextX;            // 0x24
+        uint16_t NextY;            // 0x26
+        uint8_t NextZ;             // 0x28
+        uint8_t NextFlags;         // 0x29
+        uint8_t OutsideOfPark;     // 0x2A
+        uint8_t State;             // 0x2B
+        uint8_t SubState;          // 0x2C
+        PeepSpriteType SpriteType; // 0x2D
+        uint8_t Type;              // 0x2E
```
Comment (duncanspumpkin) [inline src/openrct2/rct1/RCT1.h:450]: This is the source of your issues. Peep has two `type` variables. This PR changed it to one `Type` and one `type`.

File: src/openrct2/rct1/S4Importer.cpp [hunk 8]
```diff
@@ -2624,14 +2623,14 @@ namespace RCT1
...
                     break;
                 }
 
-                if (researchList[i].type == RCT1_RESEARCH_TYPE_RIDE)
+                if (researchList[i].Type == RCT1_RESEARCH_TYPE_RIDE)
                 {
-                    return RCT1::GetRideType(static_cast<RideType>(researchList[i].item), 0);
+                    return RCT1::GetRideType(static_cast<RideType>(researchList[i].Item), 0);
                 }
             }
```
Comment (duncanspumpkin) [inline src/openrct2/rct1/S4Importer.cpp:2636]: L2651 should be changed to `if (RCT12PeepType(static_cast<const RCT1::Peep*>(&src)->Type) == RCT12PeepType::Guest)`

File: src/openrct2/rct1/RCT1.h (hunk 7 above)
Comment (Gymnasiast) [reply src/openrct2/rct1/RCT1.h]: Thanks!

File: src/openrct2/rct1/S4Importer.cpp (hunk 8 above)
Comment (Gymnasiast) [reply src/openrct2/rct1/S4Importer.cpp:2636]: I used a different solution, one that I thought was less ambiguous.

Comment (duncanspumpkin) [reply src/openrct2/rct1/S4Importer.cpp:2636]: You may as well use the type `RCT12PeepType` for it as well. Could you check rct2's one as i think we will have the same issue.

Comment (Gymnasiast) [reply src/openrct2/rct1/S4Importer.cpp:2636]: Mercifully, RCT2’s peep has called this field `peep_type`.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**